src/main/java/org/openstreetmap/atlas/utilities/threads/Pool.java

```diff
@@ -126,11 +126,45 @@ public boolean isDead()
         return doWithTheOutput.apply(item);
     }
 
+    public <T> Result<T> queue(final Callable<T> task, final Ticker ticker)
+    {
+        final Callable<T> taskWrapper = () ->
+        {
+            try
+            {
+                return task.call();
+            }
+            finally
+            {
+                ticker.close();
+            }
+        };
+        this.queue(ticker);
+        return new Result<>(this.pool.submit(taskWrapper), this, ticker);
+    }
+
     public void queue(final Runnable command)
     {
         this.pool.execute(command);
     }
 
+    public void queue(final Runnable command, final Ticker ticker)
+    {
+        final Runnable commandWrapper = () ->
+        {
+            try
+            {
+                command.run();
+            }
+            finally
+            {
+                ticker.close();
+            }
+        };
+        this.pool.execute(ticker);
```

Comment:
One thing to note here. When using the `Pool` with a `Runnable`, it fails fast if the `Runnable` throws an exception. So you could have a case where the `ticker` throws an exception before the `queue` method even gets to the next line.

Reply:
Actually you are right, let me add ticker error handling!